lib/modules/issue/jira/index.ts

```diff
@@ -0,0 +1,281 @@
+      logger.warn({ err }, 'Could not ensure jira issue');
+    }
+    return null;
+  }
+
+  async closeIssue(issueKey: string): Promise<void> {
+    const transitions = await this.http.getJson<JiraTransitionsResponse>(
+      `/rest/api/3/issue/${issueKey}/transitions`
+    );
+
+    for (const transition of transitions.body.transitions) {
+      if (transition.to.statusCategory.key === 'done') {
+        logger.debug(`closeIssue: closing issue ${issueKey}`);
+
+        await this.http.postJson(`/rest/api/3/issue/${issueKey}/transitions`, {
+          body: {
+            transition: {
+              id: transition.id,
+            },
+          },
+        });
+
+        return;
+      }
+    }
+
+    logger.debug('Error closeIssue');
+  }
+
+  async ensureIssueClosing(title: string): Promise<void> {
+    const issues = await this.findOpenIssues(title);
+    for (const issue of issues) {
+      await this.closeIssue(issue.key);
+    }
+  }
+
+  async findOpenIssues(title: string): Promise<JiraIssue[]> {
+    const projectKey = getProjectKey();
+
+    try {
+      const jql = encodeURIComponent(
+        [
+          `summary ~ "${title}"`,
+          `project = "${projectKey}"`,
+          `reporter = currentUser()`,
+          'status != "done"', //TODO - Should we only check for `new` or `open` issues, or is not done also ok?
+        ].join(' AND ')
+      );
+
+      return (
+        (
+          await this.http.getJson<{ issues: JiraIssue[] }>(
+            `/rest/api/3/search?jql=${jql}`
+          )
+        ).body.issues || /* istanbul ignore next */ []
+      );
+    } catch (err) /* istanbul ignore next */ {
+      logger.warn({ err }, 'Error finding open jira issues');
+      return [];
+    }
+  }
+
+  massageMarkdown(input: string): string {
+    const repositoryUrl = getRepositoryUrl();
+
+    // Remove any HTML we use
+    return smartTruncate(input, 50000)
+      .replace(
+        'you tick the rebase/retry checkbox',
+        'by renaming this PR to start with "rebase!"'
+      )
+      .replace(regEx(/<\/?summary>/g), '**')
+      .replace(regEx(/<\/?(details|blockquote)>/g), '')
+      .replace(regEx(`\n---\n\n.*?<!-- rebase-check -->.*?\n`), '')
+      .replace(regEx(/\]\(\.\.\/pull\//g), `](${repositoryUrl}/pull-requests/`)
+      .replace(regEx(/<!--renovate-(?:debug|config-hash):.*?-->/g), '')
+      .replace(regEx(/\]\(\.\.\/\.\.\/pull-requests\//g), ``)
+      .replace(regEx(/\*\*\n\n\n\*\*/g), '\n#### ') // Level 4 heading for package types
+      .replace(regEx(/\n\n\*\*/g), '\n### ') // Level 3 heading for managers
+      .replace(regEx(/\*\*/g), '') // Remove closing bold tags
+      .replace(regEx(/WARN:/g), '⚠️'); // WARN to use emoji
+  }
+}
+
+/**
+ * TODO - What is the best way to retrieve these per repository to use in the Jira Issue Collector?
+ * Retrieving this may vary between platforms(ie: Bitbucket repositories can be natively linked)
+ */
+function getProjectKey(): string {
```

Comment:
I'm needing some advice on how best to configure the following for the Jira Issue Collector
-  jira project key for the repository.  
  - for bitbucket cloud, this can be fetched via a call to the /repositories/{repo}/jira/projects endpoint.  how would this value be passed down to the issue collector logic?
  - for other platform types, how would this work?
  - 
- jira cloud url.
  - similarly, this can be fetched from the bitbucket /jira/projects endpoint. 
  - for other platform types, how would this work?